### PR TITLE
Support multiple formatters

### LINKED ISSUE #530: Ability to output multiple formats
### Environment

* Elixir & Erlang/OTP versions (elixir --version): 1.15.6

* Which version of Dialyxir are you using? (cat mix.lock | grep dialyxir): 1.4.3

### Current behavior

Currently, when we run `mix dialyzer --format github`, it only outputs the annotations, resulting in a less practical CLI output (the first place you go when you have a failed build):

image

Each of our devs failed to see at first that there was now annotations in the diff view.

### Expected behavior

Ability to run `mix dialyzer --format github,dialyxir` or `mix dialyzer --format github --format dialyxir` that would output both formats.

### PR DESCRIPTION
**Pull request by axelson**
Fixes #530

One scenario where multiple formatters is helpful is on CI where you want to use the GitHub formatter along with a more verbose formatter to see more about the specific failures (especially when viewing the CI logs directly).

I add `:logger` to give the user a warning message if they provide an invalid name for a formatter but I can remove the warning if there's a specific reason to not depend on logger.

Example run:
```
> mix dialyzer --format github --format dialyxir
Finding suitable PLTs
Checking PLT...
[:compiler, :crypto, :dialyxir, :dialyzer, :elixir, :erlex, :erts, :kernel, :logger, :mix, :stdlib, :syntax_tools, :weather]
PLT is up to date!
No :ignore_warnings opt specified in mix.exs and default does not exist.

Starting Dialyzer
[
  check_plt: false,
  init_plt: ~c"/Users/jasonaxelson/dev/tmp/weather/_build/dev/dialyxir_erlang-26.2.5_elixir-1.17.2_deps-dev.plt",
  files: [~c"/Users/jasonaxelson/dev/tmp/weather/_build/dev/lib/weather/ebin/Elixir.BusinessLogic.beam",
   ~c"/Users/jasonaxelson/dev/tmp/weather/_build/dev/lib/weather/ebin/Elixir.Weather.beam",
   ~c"/Users/jasonaxelson/dev/tmp/weather/_build/dev/lib/weather/ebin/Elixir.WeatherAPI.beam",
   ~c"/Users/jasonaxelson/dev/tmp/weather/_build/dev/lib/weather/ebin/Elixir.WeatherImpl.beam"],
  warnings: [:unknown]
]
Total errors: 2, Skipped: -2, Unnecessary Skips: 0
done in 0m3.44s
::warning file=lib/weather_impl.ex,line=5,title=callback_type_mismatch::Type mismatch for @callback get_weather.
lib/weather_impl.ex:5:callback_type_mismatch
Type mismatch for @callback get_weather/1 in WeatherAPI behaviour.

Expected type:
{:error, binary()} | {:ok, map()}

Actual type:
:ok

________________________________________________________________________________
::warning file=lib/weather_impl.ex,line=11,title=callback_type_mismatch::Type mismatch for @callback fetch_weather.
lib/weather_impl.ex:11:callback_type_mismatch
Type mismatch for @callback fetch_weather/1 in WeatherAPI behaviour.

Expected type:
{:error, binary()} | {:ok, map()}

Actual type:
:ok

________________________________________________________________________________
done (warnings were emitted)
Halting VM with exit status 2
```

Updated docs in README:
Screenshot 2024-09-20 06-37-43@2x

### CODE REVIEW

**Comment by axelson** [comment]
@jeremyjh Ping! Do you have any thoughts on this PR?

File: lib/dialyxir/dialyzer.ex [hunk 1]
```diff
@@ -83,6 +69,19 @@ defmodule Dialyxir.Dialyzer do
...
+    defp parse_formatter("dialyzer"), do: Dialyxir.Formatter.Dialyzer
+    defp parse_formatter("dialyxir"), do: Dialyxir.Formatter.Dialyxir
+    defp parse_formatter("github"), do: Dialyxir.Formatter.Github
+    defp parse_formatter("ignore_file"), do: Dialyxir.Formatter.IgnoreFile
+    defp parse_formatter("ignore_file_string"), do: Dialyxir.Formatter.IgnoreFileStrict
+    defp parse_formatter("raw"), do: Dialyxir.Formatter.Raw
+    defp parse_formatter("short"), do: Dialyxir.Formatter.Short
+
+    defp parse_formatter(unknown) do
+      Logger.warning("Unrecognized formatter #{unknown} received. Falling back to default.")
```
Comment (jeremyjh) [inline lib/dialyxir/dialyzer.ex:82]: I don't have an issue with depending on `:logger`, but we already use the Mix.shell functions for text output (through Dialyxir.Output module) and Output.warning should color the text - is there a particular reason to use Logger instead ?

**Comment by jeremyjh** [comment]
@axelson Sorry, was just very busy and then forgot to get back to this. I really appreciate the contribution! Would you mind adding an update to the docs for this?

**Comment by axelson** [comment]
No worries! I've definitely taken significantly longer to respond to PRs 😅

Yeah I'll take a first crack at adding docs for this 👍

File: lib/dialyxir/dialyzer.ex (hunk 1 above)
Comment (axelson) [reply lib/dialyxir/dialyzer.ex]: Okay, updated to use `Dialyxir.Output.warning` in baaa24d722bd4694b695bcc8425e28add2cbefec

**Review by jeremyjh — APPROVED** [review]

**Comment by axelson** [comment]
Thank you! ❤️

**Pull request merged**